kgforge/specializations/models/rdf/service.py

```diff
@@ -155,9 +155,12 @@ def materialize(self, iri: URIRef) -> NodeProperties:
         """
         raise NotImplementedError()
 
-    def validate(self, resource: Resource):
+    def validate(self, resource: Resource, schema: str=None):
         try:
-            shape_iri = self.types_to_shapes[resource.type]
+            if schema is None:
+                shape_iri = self.types_to_shapes[resource.type]
+            else:
+                shape_iri = self.types_to_shapes[schema]
```

Comment:
Looking at this, it looks this implementation is expecting the value of 'schema' to be a type rather than an actual SHACL schema id. I recommend to rename it to `type` to reflect that and differentiate with `schema_id` argument in  [forge.py#L630](https://github.com/BlueBrain/nexus-forge/blob/0b04e7a1c8d6d632c8d4b2d8db6e65b79387df3d/kgforge/core/forge.py#L630) and [forge.py#L642](https://github.com/BlueBrain/nexus-forge/blob/0b04e7a1c8d6d632c8d4b2d8db6e65b79387df3d/kgforge/core/forge.py#L642).

Also a resource can have multiple types (i.e. resource.type=["type1", "type2"]), in such case the `type` argument can help the user specify which one to use. In case of multiple types and without the `type` argument provided, then a value error should be raised for now. We can extend later to support validating using multiple types.

Reply:
@MFSY  I get it. However the keyword `type` shadows the `type` built-in of Python. Is `type_id` a good alternative?

Reply:
True. Let go for `type_`. We will need later to propagate this to some forge public methods (which uses `type`) while keeping backward compatibility

Reply:
@MFSY 
> True. Let go for type_.

I pushed the change for that

> Also a resource can have multiple types (i.e. resource.type=["type1", "type2"]), in such case the type argument can help the user specify which one to use.

I don't quite understand this part. Doesn't `type_` override the `resource.type` attribute in `validate`?

Reply:
>I don't quite understand this part. Doesn't type_ override the resource.type attribute in validate?

Yes.

Reply:
My point was more about [this line](https://github.com/BlueBrain/nexus-forge/pull/217/files#diff-20fe56168eb41def80c78aa3d4c1f0dbeddb920f8859a795c86f3f148cc4c408R161) which will throw an error when resource.type is a list. A strategy (in case no type_ is provided) could be to check whether resource.type is a list and throw an exception (ValueError may be ?) with a meaningful error message.

Reply:
@MFSY I pushed a change that checks in order to raise ValueError